README.md

```diff
@@ -122,10 +185,108 @@ client.on('connected', async (conn) => {
     conn.failed();
   }
 });
+```
+
+# PulseQueue
+
+A PulseQueue declares a queue and listens for messages on that
+queue, invoking a callback for each messages.
+
+Construct a PulseQueue with the async `consume` function:
 
-client.start();
+```javascript
+let pq = await pulse.consume({
+  client,                // Client object for connecting to the server
+  bindings: [{           // exchange/routingKey patterns to bind to
+    exchange,            // Exchange to bind
+    routingKeyPattern,   // Routing key as string
+    routingKeyReference, // Reference used to parse routing keys (optional)
+  }, ..],
+  handleMessage,         // handler for incoming messages
+  queueName,             // Queue name; if omitted, set exclusiveQueue: true
+  exclusiveQueue,        // If true, use an exclusive queue (and lose messages on reconnect)
```

Comment:
I don't really understand what these snippets are suggesting -- you wrote five different things.  I can make `handleMessage` positional easily enough.  For the rest, I think most of our APIs either take options or positional args, and that mixing the two just makes it hard to remember what's what.

Reply:
Up to you... It's all ergonomics, I just wanted to give some ideas..

Since some args are required and others aren't that might be neat.. especially if it reduces indentation..

Reply:
w.r.t to your suggestion i feel making handlemessage to be made positional and such as this
`new PulseQueue(client, options, handlemessage)`